ui/src/components/chat/compose/chatbar.rs

```diff
@@ -231,6 +237,33 @@ pub fn get_chatbar<'a>(cx: &'a Scoped<'a, super::ComposeProps>) -> Element<'a> {
 
     // drives the sending of TypingIndicator
     let local_typing_ch1 = local_typing_ch.clone();
+    let enable_paste_shortcut = use_ref(cx, || true);
+
+    use_future(cx, (), |_| {
+        to_owned![enable_paste_shortcut];
+        async move {
+            loop {
+                let clipboard_data_type = check_if_there_is_file_or_string_in_clipboard()
```

Comment:
This would block the thread where the tasks are at

```suggestion
                let clipboard_data_type = tokio::task::spawn_blocking(|| {
                    check_if_there_is_file_or_string_in_clipboard()
                        .unwrap_or(ClipboardDataType::String)
                })
                .await
                .expect("Should succeed");
```